pkg/kubefed2/federate/federate.go

```diff
@@ -183,32 +185,68 @@ func GetFedResources(hostConfig *rest.Config, qualifiedTypeName, qualifiedName c
 	}, nil
 }
 
-func lookupTypeDetails(hostConfig *rest.Config, typeName ctlutil.QualifiedName) (*fedv1a1.FederatedTypeConfig, error) {
-	apiResource, err := enable.LookupAPIResource(hostConfig, typeName.Name, "")
+func getTypeConfig(hostConfig *rest.Config, typeName, namespace string, outputYAML bool) (typeconfig.Interface, error) {
+	// Lookup kubernetes API availability
+	apiResource, err := enable.LookupAPIResource(hostConfig, typeName, "")
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "Failed to find target API resource %s", typeName)
 	}
-	typeName.Name = typeconfig.GroupQualifiedName(*apiResource)
+	glog.V(2).Infof("API Resource for %s found", typeName)
 
-	client, err := genericclient.New(hostConfig)
+	typeConfig, err := attemptToLookupFederatedType(hostConfig, *apiResource, namespace)
+	if err == nil {
+		return typeConfig, nil
+	}
+	if apierrors.IsNotFound(err) && !outputYAML {
+		return nil, errors.Errorf("%v. Try 'kubefed2 enable type %s' before federating the resource", err, typeName)
+	}
+	if outputYAML {
+		glog.V(1).Infof("Lookup of installed type failed: %v - Falling back on generated default values instead.", err)
+		return enable.GenerateTypeConfigForTarget(*apiResource, enable.NewEnableTypeDirective()), nil
+	}
+
+	return nil, err
+}
+
+func attemptToLookupFederatedType(hostConfig *rest.Config, apiResource metav1.APIResource, namespace string) (typeconfig.Interface, error) {
+	resolvedTypeName := typeconfig.GroupQualifiedName(apiResource)
+	// Lookup federated API availability
+	federatedTypeName := federatedTypeName(apiResource)
+	_, err := enable.LookupAPIResource(hostConfig, federatedTypeName, "")
 	if err != nil {
-		return nil, errors.Wrap(err, "Failed to get federation client")
+		return nil, err
+	}
+	if err == nil {
```

Comment:
This conditional is redundant, since by definition anything after the `err != nil` block is the case where `err == nil`.

Reply:
Silly miss, updated!